### ISSUE TITLE
Implement IndexedDB API

### ISSUE THREAD

**Issue by spaceface777** [HermesMessenger/Hermes#117]
The [IndexedDB API](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) is a database inside the browser that can store text, JSON, and binary blobs. It can be used inside the main page as well as web workers and service workers.

Off the top of my head I can think of a few current issues that this API would solve completely:
  - Saving messages locally so that a user can see them if they are offline (this also requires a service worker because we also need to cache the HTML, JS and CSS for the page)
  - Saving user settings locally so that when loading the page we can display a user's previous settings instead of the generic ones

**Comment by ThePerkinrex** [comment]
Yeah, and for storing blobs(profile pictures) it should be more optimised than just a js string
